tests/functional/test_receipt.py

```diff
@@ -116,12 +116,22 @@ def test_decode_logs_unspecified_abi_gets_all_logs(owner, contract_instance):
 def test_get_failed_receipt(owner, vyper_contract_instance, eth_tester_provider):
     # Setting to '5' always fails.
     transaction = vyper_contract_instance.setNumber.as_transaction(
-        5, sender=owner, gas_limit=100000
+        5, sender=owner, gas_limit=100000, _raise_on_fail=False
```

Comment:
Specifically here is where the field doesn't work if it's not Pydantic. Not that it necessarily has to be, but it seems like it means updating all the functions that create a serialization from `**kwargs`?